lib/elixir/lib/calendar/iso.ex

```diff
@@ -980,9 +1036,10 @@ defmodule Calendar.ISO do
   @impl true
   @spec valid_time?(Calendar.hour(), Calendar.minute(), Calendar.second(), Calendar.microsecond()) ::
           boolean
-  def valid_time?(hour, minute, second, {microsecond, precision}) do
-    hour in 0..23 and minute in 0..59 and second in 0..59 and microsecond in 0..999_999 and
-      precision in 0..6
+  def valid_time?(hour, minute, second, microsecond)
+      when is_integer(hour) and hour >= 0 and is_integer(minute) and minute >= 0 and
```

Comment:
Let's only keep the type check in guards. `hour >= 0` should be moved to the body. No need for `is_microsecond` here too, let's pattern match and assert the contents without raising.

Reply:
done. please have a look